src/renderers/native/ReactNativeComponentTree.js

```diff
@@ -39,28 +39,41 @@ function precacheNode(inst, tag) {
   instanceCache[tag] = nativeInst;
 }
 
+function precacheFiberNode(hostInst, tag) {
+  instanceCache[tag] = hostInst;
+}
+
 function uncacheNode(inst) {
-  var tag = inst._rootNodeID;
+  // TODO (bvaughn) Clean up once Stack is deprecated
```

Comment:
Don't you always know at the caller whether this is stack or Fiber? If you're using this function still, let's split it up into two.